File: src/main/java/com/flexpoker/chat/repository/ChatRepository.java
package com.flexpoker.chat.repository;

import java.util.List;
import java.util.UUID;

import com.flexpoker.web.dto.OutgoingChatMessageDTO;

public interface ChatRepository {

    void saveChatMessage(OutgoingChatMessageDTO chatMessage);

    List<OutgoingChatMessageDTO> fetchAllLobbyChatMessages();

    List<OutgoingChatMessageDTO> fetchAllGameChatMessages(UUID gameId);

    List<OutgoingChatMessageDTO> fetchAllTableChatMessages(UUID tableId);

}

File: src/main/kotlin/com/flexpoker/chat/repository/ChatRepository.kt
package com.flexpoker.chat.repository

import com.flexpoker.web.dto.OutgoingChatMessageDTO
import java.util.UUID

interface ChatRepository {
    fun saveChatMessage(chatMessage: OutgoingChatMessageDTO)
    fun fetchAllLobbyChatMessages(): List<OutgoingChatMessageDTO>
    fun fetchAllGameChatMessages(gameId: UUID): List<OutgoingChatMessageDTO>
    fun fetchAllTableChatMessages(tableId: UUID): List<OutgoingChatMessageDTO>
}